### DISCUSSION TITLE
Is there any other well known socket.io parsers except from default and msgpack ?

### DISCUSSION THREAD

**Discussion by Totodore** [socketio/socket.io#4934] [General] [answered]
I want to implement custom [parsers support for my rust impl of socket.io](https://github.com/Totodore/socketioxide/issues/234). However my implementation depends if I need to organize that as external libraries (like the official implementation). Or if I simply propose an internal implementation for both default and msgpack? 

@darrachequesne any thoughts on this? Did you see a lot of custom socket.io parsers implementations?

**Comment by darrachequesne** [comment, marked as answer]
The default parser and the one based on MessagePack are indeed the most used parsers.

There are also:

- the JSON parser: https://www.npmjs.com/package/socket.io-json-parser
- the cbor-x parser: https://www.npmjs.com/package/socket.io-cbor-x-parser

But the number of downloads is anecdotal.

**Reply by Totodore** [reply]
Thanks :)
